### PR TITLE
Clean up internal library dependencies so we can avoid building unused code

### PR DESCRIPTION
**Pull request by luke-jr**
Without this, we unnecessarily compile libbitcoin_server for utils-only configurations.

### CODE REVIEW

**Comment by laanwj** [comment]
@theuni can you take a look here?

**Comment by theuni** [comment]
Looking into this.

**Comment by luke-jr** [comment]
Rewrote based on @theuni 's version (nice!), but also fixed configure to tolerate missing boost for lib-only builds, and check for openssl/ec.h (missing on Gentoo if OpenSSL is built with "bindist" USE flag).

File: configure.ac [hunk 1]
```diff
@@ -500,6 +520,14 @@ if test x$use_upnp != xno; then
   )
 fi
 
+if test x$build_bitcoin_utils$build_bitcoind$bitcoin_enable_qt$use_tests = xnonono; then
```
Comment (theuni) [inline configure.ac]: BITCOIN_QT_CONFIGURE needs to be run before we know about bitcoin_enable_qt.

Comment (luke-jr) [inline configure.ac]: fixed

**Comment by luke-jr** [comment]
Added another bugfix and retested here (incl boost-less and ec-less systems). I think this is finally done.

**Comment by theuni** [comment]
Thanks for this! ut ACK if travis is happy.

**Pull request merged**